drivers/net/wireless/mediatek/mt76/mt7921/mac.c

```diff
@@ -1301,24 +1302,30 @@ mt7921_mac_reset(struct mt7921_dev *dev)
 /* system error recovery */
 void mt7921_mac_reset_work(struct work_struct *work)
 {
-	struct ieee80211_hw *hw;
-	struct mt7921_dev *dev;
+	struct mt7921_dev *dev = container_of(work, struct mt7921_dev,
+					      reset_work);
+	struct ieee80211_hw *hw = mt76_hw(dev);
+	struct mt76_connac_pm *pm = &dev->pm;
 	int i;
 
-	dev = container_of(work, struct mt7921_dev, reset_work);
-	hw = mt76_hw(dev);
-
 	dev_err(dev->mt76.dev, "chip reset\n");
 	ieee80211_stop_queues(hw);
 
 	cancel_delayed_work_sync(&dev->mphy.mac_work);
-	cancel_delayed_work_sync(&dev->pm.ps_work);
-	cancel_work_sync(&dev->pm.wake_work);
 	cancel_work_sync(&dev->phy.roc.work);
 	del_timer_sync(&dev->phy.roc.timer);
 
+	set_bit(MT76_RESET, &dev->mphy.state);
+	cancel_delayed_work_sync(&pm->ps_work);
+	cancel_work_sync(&pm->wake_work);
+
 	mutex_lock(&dev->mt76.mutex);
 	for (i = 0; i < 10; i++) {
+		__mt7921_mcu_drv_pmctrl(dev);
+		clear_bit(MT76_STATE_PM, &dev->mphy.state);
+		dev->pm.stats.last_wake_event = jiffies;
+		dev->pm.stats.last_doze_event = jiffies;
+
```

Comment:
@objelf why not moving dev->pm.stats.last_wake_event/dev->pm.stats.last_doze_event initialization and clear_bit() inside __mt7921_mcu_drv_pmctrl()?

Reply:
it seems doable. let's do it in the next version.